eventarc/storage_handler/main.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2023 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START eventarc_storage_cloudevent_server]
+import os
+
+from cloudevents.http import from_http
+
+from flask import Flask, request
+
+from google.events.cloud.storage import StorageObjectData
+
+
+app = Flask(__name__)
+# [END eventarc_storage_cloudevent_server]
+
+
+# [START eventarc_storage_cloudevent_handler]
+@app.route("/", methods=["POST"])
+def index():
+    event = from_http(request.headers, request.get_data())
+
+    # Gets the GCS bucket name from the CloudEvent data
+    # Example: "storage.googleapis.com/projects/_/buckets/my-bucket"
+    storage_obj = StorageObjectData(event.data)
+    gcs_object = os.path.join(storage_obj.bucket, storage_obj.name)
+    update_time = storage_obj.updated
```

Comment:
This is using the timestamp from the Cloud Storage object.
In the plan, we had the intention to surface the date from the CloudEvent envelop instead. The [(optional) time property](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/spec.md#time) in that case.

Reply:
per product guidance, the time property is present, but may not strictly align with the GCS modification time.